### PR TITLE
Remove unshipped assembly versions

### PR DESCRIPTION
**Pull request by ericstj**
These are from packages that never shipped stable and we've since moved
past these versions.

/cc @weshaggard @chcosta

### CODE REVIEW

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 1]
```diff
@@ -1008,7 +998,6 @@
       "AssemblyVersionInPackageVersion": {
         "4.0.0.0": "4.0.0",
         "4.0.10.0": "4.0.10",
-        "4.1.0.0": "4.0.12",
         "4.0.11.0": "4.0.11"
```
Comment (chcosta) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: Are we no longer updating System.Globalization?

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 2]
```diff
@@ -1424,7 +1407,6 @@
       "AssemblyVersionInPackageVersion": {
         "4.0.0.0": "4.0.0",
         "4.0.10.0": "4.0.10",
-        "4.0.11.0": "4.0.11",
         "4.0.12.0": "4.0.12"
```
Comment (chcosta) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: Same question as for System.Globalization

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 3]
```diff
@@ -1545,11 +1527,9 @@
       "AssemblyVersionInPackageVersion": {
         "1.0.22.0": "1.0.22",
```
Comment (chcosta) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: We also shipped a 1.0.21

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json (hunk 1 above)
Comment (ericstj) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: This was inconsistent with the actual repo state.  Assembly version 4.1.0.0 does not currently build, even if it did it should never be put in a package versioned 4.0.12.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json (hunk 2 above)
Comment (ericstj) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: This was inconsistent with the repo state.  We never shipped this version stable and we've moved past it.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json (hunk 3 above)
Comment (ericstj) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: This PR is not adding additional versions, just deleting versions that were "dead".  The 1.0.21 version is not currently required by anything in the repo, should it be in the future that would generate an error and we could add it.

**Pull request merged**